hrreport/CRM/HRReport/Upgrader.php

```diff
@@ -247,21 +247,45 @@ public function upgrade_1404() {
       'CRM_HRReport_Form_Activity_HRAbsenceDates'
     ];
     $reports = civicrm_api3('OptionValue', 'get', [
-      'return' => ['name'],
+      'return' => ['label'],
       'option_group_id' => 'report_template',
       'name' => ['IN' => $templates],
     ]);
 
     if ($reports['count'] > 0) {
       $templateIds = array_column(array_values($reports['values']), 'id');
+      $templateLabels = array_column(array_values($reports['values']), 'label');
       foreach ($templateIds as $templateId) {
         civicrm_api3('OptionValue', 'delete', ['id' => $templateId]);
       }
 
-      $sql = "DELETE FROM `civicrm_managed` WHERE `entity_id` IN ('" . implode("','", $templateIds) . "')";
+      $sql = "DELETE FROM `civicrm_managed` WHERE `entity_type` = 'ReportTemplate'";
+      $sql .= " AND `name` IN ('" . implode(" Template','", $templateLabels) . " Template')";
```

Comment:
why did you change this to delete reports based on the labels instead of IDs? The only change necessary here was to add `entity_type = 'ReportTemplate'` to the query criteria

Reply:
That must have been a misinterpretation 🤷‍♂️ .  More so, the report instances was using names.  Will revert back to id with entity_type clause. ♻️